File: Week 3/Gambler/src/main/java/edu/atudonegal/Gambler.java
package edu.atudonegal;

import java.io.IOException;
import java.util.Random;
import java.util.Scanner;
import java.util.concurrent.ThreadLocalRandom;

/**
 * The class Gambler provides a simple implementation of Gambler's ruin. The code starts by asking
 * the player for their starting pot, then their target goal. The game should then simulate the
 * throwing of two dice. If the value of the dice thrown is <6 then the player wins €1, otherwise
 * they loose €1. The game continues until the value of their pot = goal, or they lose all of their
 * money. After each game the player should be asked if they would like to play again. If they do,
 * then they should be asked once again for their starting pot and goal. For more information see:
 * https://en.wikipedia.org/wiki/Gambler%27s_ruin
 */
public class Gambler {

  /**
   * The main entry point of the application.
   *
   * @param args program input arguments - not used.
   */
  public static void main(String[] args) {

    int bets = 0;
    int gameWins = 0;
    int winningThrows = 0;

    String playAgainResponse = "Y";

    Scanner scan = new Scanner(System.in);

    while (playAgainResponse.equals("Y")) {
      int pot = getPot();
      int goal = getGoal();

      // Game logic
      while (pot > 0 && pot < goal) {
        // TO DO - Implement game logic
         bets++;
         if(Math.random() < 0.5)
         { 
            pot++;
         }
         else
         {
            pot--;
         }
         
      }
      if (pot == goal) {
        gameWins++;
        System.out.print(
            "\n\nCongratulations, you beat Gambler's Ruin. Would you like to try your luck again?"
                + " y/n: ");
        pot = 0;

      } else {
        System.out.print("\n\nToo bad, you lost to Gambler's Ruin. "
            + "Would you like to try your luck again? y/n: ");
      }
      String temp = scan.nextLine().toUpperCase();
      playAgainResponse = temp;
    }

    System.out.println("Thank you for playing Gambler's Ruin");
    System.out.println("\n\nYour game statistics are:");
    System.out.println("Number of throws: " + bets);
    System.out.println("Number of winning throws: " + winningThrows);
    System.out.println("Percentage wins : " + 100.0 * winningThrows / bets);
    System.out.println("Winning games : " + gameWins);


  }

  /**
   * Returns an amount of money the player starts with in their pot.
   *
   * @return the amount of money the player starts the game with
   */
  public static int getPot() {

    Scanner scan = new Scanner(System.in);
    System.out.print("How much is your starting bankroll? €: ");
    return scan.nextInt();
  }

  /**
   * Returns an amount of money the player would like to win.
   *
   * @return the amount of money they player would like to win
   */
  public static int getGoal() {
    Scanner scan = new Scanner(System.in);
    System.out.print("How much is your starting bankroll? €: ");
    return scan.nextInt();
  }

  /**
   * Returns the value of two dice being thrown.
   *
   * @return a number in the range [2..12] the result of throwing to dice
   */
  public static int simulateThrow() {

    // TODO
    //Random randomNumber = new Random();
    int randomNum = ThreadLocalRandom.current().nextInt(2, 12 + 1);
    //return randomNumber.nextInt(13);
    return randomNum;
  }
}

File: Week 7/Code style demo/VS/GamblersRuin/GamblersRuin/Gambler.cs
// Written by Dr. Shane Wilson.
// The author licenses this file to you under the MIT license.
// See the LICENSE file in the solution root for more information.

using System;
using System.Collections.Generic;
using System.Linq;
using System.Numerics;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Serialization;
using Microsoft.VisualBasic;
using static System.Net.WebRequestMethods;

namespace GamblersRuin
{
    /// <summary>
    /// This example of Gambler's ruin has been written incorrectly to demonstrate code analysis and formatting tools within visual studio. This line is therefore excessively long for a reason!
    /// 
    /// The class Gambler provides a simple implementation of Gambler's ruin. The code starts by asking
    /// the player for their starting pot, then their target goal.The game should then simulate the
    ///throwing of two dice.If the value of the dice thrown is <6 then the player wins €1, otherwise
    /// they loose €1. The game continues until the value of their pot = goal, or they lose all of their
    ///money.After each game the player should be asked if they would like to play again. If they do,
    ///then they should be asked once again for their starting pot and goal.For more information see:
    ///https://en.wikipedia.org/wiki/Gambler%27s_ruin
    /// </summary>
    public static class Gambler {

        private static int bets = 0;
        static private int gameWins = 0;
        private static int winningThrows = 0;
        static private string play_again_response = "Y";

        public static void play_Game() {
            while (play_again_response == "Y") { 
            int pot = getPot();
            int goal = getGoal();
               while (pot > 0 && pot < goal) {
                    //gameplay
                    bets = bets+ 1;
                    int roll = rolldice();
                    Console.WriteLine("\nThrowing the dice....");
                    Console.WriteLine("You threw a:  " + roll);

                    if(roll< 6)                    {
                        pot = pot + 1;
                        winningThrows = winningThrows+ 1;
                        Console.WriteLine($"Congratulations you won 1 and you now have: {pot}");
                    }
                    else {
                        pot = pot - 1;
                        Console.WriteLine($"Bad luck, you lost 1 and you now have: {pot}");
                    }

                }
            if (pot == goal) {
                gameWins++;
                Console.Write(
                    "\n\nCongratulations, you beat Gambler's Ruin. Would you like to try your luck again?"
                        + " y/n: ");
                pot = 0;

            } else
            {
                Console.Write("\n\nToo bad, you lost to Gambler's Ruin. "
                    + "Would you like to try your luck again? y/n: ");
            }
            var response = Console.ReadLine()!.ToUpper();
            play_again_response = response;
        }
            printStats();        }

        static void printStats()
        {
            Console.WriteLine("Thank you for playing Gambler's Ruin");
            Console.WriteLine("\n\nYour game statistics are:");
            Console.WriteLine("Number of throws: " + bets);
            Console.WriteLine("Number of winning throws: " + winningThrows);
            Console.WriteLine("Percentage wins : " + 100.0 * winningThrows / bets);
            Console.WriteLine("Winning games : " + gameWins);
        }
        

     static int getPot() {

            Console.Write("How what is your opening pot:");
            var openingPot = Convert.ToInt16(Console.ReadLine());
            return openingPot;
        }

        static int getGoal() {

            Console.Write("What is your goal amount:");
            var goal = Convert.ToInt16(Console.ReadLine());
            return goal;
        }

        static int rolldice()    {
            Random rand = new Random();
            return rand.Next(2, 12);
        }

    } }


